### PR TITLE
ADD: Doc/java doc

### LINKED ISSUE #169: Package description
We could think of adding package description for java doc.

### PR DESCRIPTION
**Pull request by Segelzwerg**
closes #169

### CODE REVIEW

File: src/main/java/mops/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/**
+ Modularen Online Plattform für Studierende.
```
Comment (kosch104) [inline src/main/java/mops/package-info.java]:
```suggestion
 Modulare Online Plattform für Studierende.
```

**Comment by Segelzwerg** [comment]
Not done yet!

File: src/main/java/mops/persistence/directory/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
+Metadas of directories database integration.
```
Comment (kosch104) [inline src/main/java/mops/persistence/directory/package-info.java]:
```suggestion
Metadata of directories database integration.
```
?

**Comment by Segelzwerg** [comment]
Ready for review.

File: src/main/java/mops/persistence/directory/DirectoryBuilder.java [hunk 3]
```diff
@@ -49,12 +52,12 @@
...
-        this.parentId = directory.getParentId();
-        this.groupOwner = directory.getGroupOwner();
-        this.permissionsId = directory.getPermissionsId();
-        this.creationTime = directory.getCreationTime();
+        id = directory.getId();
+        name = directory.getName();
+        parentId = directory.getParentId();
+        groupOwner = directory.getGroupOwner();
+        permissionsId = directory.getPermissionsId();
+        creationTime = directory.getCreationTime();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/directory/DirectoryBuilder.java]:
```suggestion
        this.id = directory.getId();
        this.name = directory.getName();
        this.parentId = directory.getParentId();
        this.groupOwner = directory.getGroupOwner();
        this.permissionsId = directory.getPermissionsId();
        this.creationTime = directory.getCreationTime();
```

File: src/main/java/mops/persistence/directory/DirectoryBuilder.java [hunk 4]
```diff
@@ -65,9 +68,9 @@ public DirectoryBuilder from(@NonNull Directory directory) {
      * @return this
      */
     public DirectoryBuilder fromParent(@NonNull Directory parent) {
-        this.parentId = parent.getId();
-        this.groupOwner = parent.getGroupOwner();
-        this.permissionsId = parent.getPermissionsId();
+        parentId = parent.getId();
+        groupOwner = parent.getGroupOwner();
+        permissionsId = parent.getPermissionsId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/directory/DirectoryBuilder.java]:
```suggestion
        this.parentId = parent.getId();
        this.groupOwner = parent.getGroupOwner();
        this.permissionsId = parent.getPermissionsId();
```

File: src/main/java/mops/persistence/directory/DirectoryBuilder.java [hunk 5]
```diff
@@ -89,7 +92,7 @@ public DirectoryBuilder id(Long id) {
      * @return this
      */
     public DirectoryBuilder id(Directory directory) {
-        this.id = directory == null ? null : directory.getId();
+        id = directory == null ? null : directory.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/directory/DirectoryBuilder.java]:
```suggestion
        this.id = directory == null ? null : directory.getId();
```

File: src/main/java/mops/persistence/directory/DirectoryBuilder.java [hunk 6]
```diff
@@ -122,7 +125,7 @@ public DirectoryBuilder parent(long parentId) {
      * @return this
      */
     public DirectoryBuilder parent(@NonNull Directory parent) {
-        this.parentId = parent.getId();
+        parentId = parent.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/directory/DirectoryBuilder.java]:
```suggestion
        this.parentId = parent.getId();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 7]
```diff
@@ -61,14 +64,14 @@
...
-        this.directoryId = file.getDirectoryId();
-        this.type = file.getType();
-        this.size = file.getSize();
-        this.owner = file.getOwner();
+        id = file.getId();
+        name = file.getName();
+        directoryId = file.getDirectoryId();
+        type = file.getType();
+        size = file.getSize();
+        owner = file.getOwner();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        this.id = file.getId();
        this.name = file.getName();
        this.directoryId = file.getDirectoryId();
        this.type = file.getType();
        this.size = file.getSize();
        this.owner = file.getOwner();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 8]
```diff
@@ -79,9 +82,9 @@ public FileInfoBuilder from(@NonNull FileInfo file) {
      * @return this
      */
     public FileInfoBuilder from(@NonNull MultipartFile file) {
-        this.name = file.getName();
-        this.type = file.getContentType();
-        this.size = file.getSize();
+        name = file.getName();
+        type = file.getContentType();
+        size = file.getSize();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        this.name = file.getName();
        this.type = file.getContentType();
        this.size = file.getSize();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 9]
```diff
@@ -61,14 +64,14 @@
...
-        this.owner = file.getOwner();
+        id = file.getId();
+        name = file.getName();
+        directoryId = file.getDirectoryId();
+        type = file.getType();
+        size = file.getSize();
+        owner = file.getOwner();
         file.getTags().stream().map(FileTag::getName).forEach(this::tag);
-        this.creationTime = file.getCreationTime();
+        creationTime = file.getCreationTime();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        this.creationTime = file.getCreationTime();
```

File: src/main/java/mops/persistence/directory/DirectoryBuilder.java [hunk 10]
```diff
@@ -155,7 +158,7 @@ public DirectoryBuilder permissions(long permissionsId) {
      * @return this
      */
     public DirectoryBuilder permissions(@NonNull DirectoryPermissions permissions) {
-        this.permissionsId = permissions.getId();
+        permissionsId = permissions.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/directory/DirectoryBuilder.java]:
```suggestion
        this.permissionsId = permissions.getId();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 11]
```diff
@@ -103,7 +106,7 @@ public FileInfoBuilder id(Long id) {
      * @return this
      */
     public FileInfoBuilder id(FileInfo fileInfo) {
-        this.id = fileInfo == null ? null : fileInfo.getId();
+        id = fileInfo == null ? null : fileInfo.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        id = fileInfo == null ? null : fileInfo.getId();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 12]
```diff
@@ -140,7 +143,7 @@ public FileInfoBuilder directory(long directoryId) {
      * @return this
      */
     public FileInfoBuilder directory(@NonNull Directory directory) {
-        this.directoryId = directory.getId();
+        directoryId = directory.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        this.directoryId = directory.getId();
```

File: src/main/java/mops/persistence/file/FileInfoBuilder.java [hunk 13]
```diff
@@ -196,7 +199,7 @@ public FileInfoBuilder tag(@NonNull String tag) {
             log.error("Failed to add tag as it was empty.");
             throw new IllegalArgumentException("tag must not be empty!");
         }
-        this.tags.add(new FileTag(tag));
+        tags.add(new FileTag(tag));
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/file/FileInfoBuilder.java]:
```suggestion
        this.tags.add(new FileTag(tag));
```

File: src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java [hunk 14]
```diff
@@ -36,9 +39,9 @@
      * @return this
      */
     public DirectoryPermissionsBuilder from(DirectoryPermissions permissions) {
-        this.id = permissions.getId();
+        id = permissions.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java]:
```suggestion
        this.id = permissions.getId();
```

File: src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java [hunk 15]
```diff
@@ -36,9 +39,9 @@
...
         permissions.getPermissions().forEach(e -> entry(e.getRole(), e.isCanRead(), e.isCanWrite(), e.isCanDelete()));
-        this.creationTime = permissions.getCreationTime();
+        creationTime = permissions.getCreationTime();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java]:
```suggestion
        this.creationTime = permissions.getCreationTime();
```

File: src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java [hunk 16]
```diff
@@ -60,7 +63,7 @@ public DirectoryPermissionsBuilder id(Long id) {
      * @return this
      */
     public DirectoryPermissionsBuilder id(DirectoryPermissions permissions) {
-        this.id = permissions == null ? null : permissions.getId();
+        id = permissions == null ? null : permissions.getId();
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java]:
```suggestion
        this.id = permissions == null ? null : permissions.getId();
```

File: src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java [hunk 17]
```diff
@@ -78,7 +81,7 @@ public DirectoryPermissionsBuilder entry(@NonNull String role, boolean canRead,
         if (role.isEmpty()) {
             throw new IllegalArgumentException("role must not be empty!");
         }
-        this.entries.add(new DirectoryPermissionEntry(role, canRead, canWrite, canDelete));
+        entries.add(new DirectoryPermissionEntry(role, canRead, canWrite, canDelete));
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/permission/DirectoryPermissionsBuilder.java]:
```suggestion
        this.entries.add(new DirectoryPermissionEntry(role, canRead, canWrite, canDelete));
```

File: src/main/java/mops/persistence/permission/package-info.java [hunk 18]
```diff
@@ -0,0 +1,4 @@
+/**
+Directory Permission database integration.
```
Comment (iTitus) [inline src/main/java/mops/persistence/permission/package-info.java]:
```suggestion
 * Directory Permission database integration.
```

File: src/main/java/mops/persistence/package-info.java [hunk 19]
```diff
@@ -0,0 +1,4 @@
+/**
+Persistence layer.
```
Comment (iTitus) [inline src/main/java/mops/persistence/package-info.java]:
```suggestion
 * Persistence layer.
```

File: src/main/java/mops/persistence/file/package-info.java [hunk 20]
```diff
@@ -0,0 +1,4 @@
+/**
+Metadata of files database integration.
```
Comment (iTitus) [inline src/main/java/mops/persistence/file/package-info.java]:
```suggestion
 * Metadata of files database integration.
```

File: src/main/java/mops/persistence/directory/package-info.java [hunk 21]
```diff
@@ -0,0 +1,4 @@
+/**
+Metadata of directories database integration.
```
Comment (iTitus) [inline src/main/java/mops/persistence/directory/package-info.java]:
```suggestion
 * Metadata of directories database integration.
```

File: src/main/java/mops/persistence/config/package-info.java [hunk 22]
```diff
@@ -0,0 +1,4 @@
+/**
+Persistence layer configuration.
```
Comment (iTitus) [inline src/main/java/mops/persistence/config/package-info.java]:
```suggestion
 * Persistence layer configuration.
```

File: src/main/java/mops/package-info.java [hunk 23]
```diff
@@ -0,0 +1,4 @@
+/**
+ Modulare Online Plattform für Studierende.
```
Comment (iTitus) [inline src/main/java/mops/package-info.java]:
```suggestion
  * Modulare Online Plattform für Studierende.
```

File: src/main/java/mops/exception/package-info.java [hunk 24]
```diff
@@ -0,0 +1,4 @@
+/**
+Base Exceptions.
```
Comment (iTitus) [inline src/main/java/mops/exception/package-info.java]:
```suggestion
 * Base Exceptions.
```

File: src/main/java/mops/config/package-info.java [hunk 25]
```diff
@@ -0,0 +1,4 @@
+/**
+ Configuration of the application.
```
Comment (iTitus) [inline src/main/java/mops/config/package-info.java]:
```suggestion
 *  Configuration of the application.
```

File: src/main/java/mops/businesslogic/utils/package-info.java [hunk 26]
```diff
@@ -0,0 +1,4 @@
+/**
+Statics in business layer.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/utils/package-info.java]:
```suggestion
 * Statics in business layer.
```

File: src/main/java/mops/businesslogic/query/package-info.java [hunk 27]
```diff
@@ -0,0 +1,4 @@
+/**
+File search tools.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/query/package-info.java]:
```suggestion
 * File search tools.
```

File: src/main/java/mops/businesslogic/prometheus/package-info.java [hunk 28]
```diff
@@ -0,0 +1,4 @@
+/**
+Prometheus integration.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/prometheus/package-info.java]:
```suggestion
 * Prometheus integration.
```

File: src/main/java/mops/businesslogic/package-info.java [hunk 29]
```diff
@@ -0,0 +1,4 @@
+/**
+ Business Layer contains all services.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/package-info.java]:
```suggestion
 * Business Layer contains all services.
```

File: src/main/java/mops/businesslogic/exception/package-info.java [hunk 30]
```diff
@@ -0,0 +1,4 @@
+/**
+Exceptions thrown from business layer.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/exception/package-info.java]:
```suggestion
 * Exceptions thrown from business layer.
```

File: src/main/java/mops/businesslogic/config/package-info.java [hunk 31]
```diff
@@ -0,0 +1,4 @@
+/**
+ Configuration for services.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/config/package-info.java]:
```suggestion
  * Configuration for services.
```

File: src/main/java/mops/presentation/package-info.java [hunk 32]
```diff
@@ -0,0 +1,4 @@
+/**
+Presentation layer: contains all Controllers.
```
Comment (iTitus) [inline src/main/java/mops/presentation/package-info.java]:
```suggestion
 * Presentation layer: contains all Controllers.
```

File: src/main/java/mops/utils/package-info.java [hunk 33]
```diff
@@ -0,0 +1,4 @@
+/**
+General static classes.
```
Comment (iTitus) [inline src/main/java/mops/utils/package-info.java]:
```suggestion
 * General static classes.
```

File: src/main/java/mops/businesslogic/config/package-info.java [hunk 34]
```diff
@@ -0,0 +1,4 @@
+/**
+  * Configuration for services.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/config/package-info.java]:
```suggestion
 * Configuration for services.
```

**Pull request merged**